### PR TITLE
Bundle JS Files

### PR DESCRIPTION
**Pull request by ArquintL**
Uses webpack as suggested by Microsoft to reduce the number of JS files in the packaged extension.
In addition, the following improvements are included:
- client & server output is no longer written to `console` or a log file but 2 VSCode channels are used instead
- caching of SBT has been improved: there won't be any cache hits on the primary key (but on the restore keys) which means that an updated version will be uploaded to the cache after compilation (there's no cache update if hit on primary key occurs)
- npm dependencies have been updated and unused one removed
- extension packaging has been fixed/improved to only include required files and to correctly link to images

### CODE REVIEW

File: client/package-notes.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+Be aware of strange behavior during testing when using the following setup:
+- Webpack is used to bundle the extension as `dist/extension.js`.
+- `package.json` is configured accordingly to use the bundled extension.
+- `tsc` is invoked to transpile all typescript files to `out/` and the extensionTestsPath is configured to be `out/test/index.js` i.e. run the tests.
+
+In this particular setup, the tests will trigger the extension's activation and VSCode will start the extension from `dist/extension.js` (as configured in `package.json`).
+However, the test files do not have access to the extension's global state as the tests import the extension files that are located in `out/` (instead of the actually run extension in `dist/`).
+Hence, the tests will access global state of an extension (the one in `out/`) that is never started by VSCode.
+
+There might be a workaround for this setup by not importing extension files into the tests but use the following snippet and `extensionApi` to access the extension's public API:
```
Comment (Felalolf) [inline client/package-notes.md:25]: "into the tests but" -> "into the tests, but"

File: client/package-notes.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Gobra IDE Build System
```
Comment (Felalolf) [inline client/package-notes.md:1]: For whom is this file? For us developers or can IDE users also see this?

**Review by Felalolf — APPROVED** [review]
As always, I do not really know what is going on, but your descriptions seem good.

File: client/package-notes.md (hunk 2 above)
Comment (ArquintL) [reply client/package-notes.md:1]: Just for us. As the README will be visible to IDE users, I've had to create a different file for that

**Pull request merged**